Decouple iOS Error Tracking docs from RUM

rtrieu opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->
- Copies iOS SDK setup instructions from RUM to Error Tracking, with lots of rewriting and rearranging of content
- Removes mentions of Error Tracking from the RUM Android SDK setup steps except where relevant
- Reorders links

### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

**For Datadog employees**:

Your branch name MUST follow the `<name>/<description>` convention and include the forward slash (`/`). Without this format, your pull request will not pass CI, the GitLab pipeline will not run, and you won't get a branch preview. Getting a branch preview makes it easier for us to check any issues with your PR, such as broken links.

If your branch doesn't follow this format, rename it or create a new branch and PR.

[6/5/2025] Merge queue has been disabled on the documentation repo. If you have write access to the repo, the PR has been reviewed by a Documentation team member, and all of the required checks have passed, you can use the **Squash and Merge** button to merge the PR. If you don't have write access, or you need help, reach out in the #documentation channel in Slack.

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

janine-c commented on content/en/error_tracking/frontend/mobile/ios.md:
```diff
@@ -1,8 +1,749 @@
+
+## Limitations
+
+- See [Supported versions][20] for a list operating system versions and platforms that are compatible with the iOS SDK.
```
I would move this information to the beginning. It feels odd to have a compatibility list after all this info - I think users are likely to look for this info when they're trying to figure out whether they can use a feature to solve a problem they're facing.

rtrieu commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/setup.md:
```diff
@@ -362,34 +353,36 @@ configuration.site = [DDSite ap2];
 {{< /site-region >}}
 
-The iOS SDK automatically tracks user sessions depending on options provided at the SDK initialization. To add GDPR compliance for your EU users and other [initialization parameters][6] to the SDK configuration, see the [Set tracking consent documentation](#set-tracking-consent-gdpr-compliance).
+The iOS SDK automatically tracks user sessions based on the options you provide during SDK initialization. To add GDPR compliance for your EU users (required for apps targeting European users) and configure other [initialization parameters][5], see the [Set tracking consent documentation](#set-tracking-consent-gdpr-compliance).
```
yea, this will have to be in a different PR! noted.

rtrieu commented:
Hi @janine-c, I appreciate the thorough review that helped move rearrange some of the content so that limitations are described up front and info was consolidated where it made sense. I think I addressed all your feedback. Can you take another look when you get the chance?

janine-c commented on content/en/error_tracking/frontend/mobile/ios.md:
```diff
@@ -1,8 +1,746 @@
+
+Crash reporting captures fatal crashes when your app terminates unexpectedly, in addition to the errors that Error Tracking displays in a unified interface.
+
+To enable crash reporting, add the package according to your dependency manager and update your initialize snippet.
```
```suggestion
To enable crash reporting, add the package according to your dependency manager and update your initialization snippet.
```

janine-c commented on content/en/error_tracking/frontend/mobile/ios.md:
```diff
@@ -1,8 +1,746 @@
+
+### Step 5 - Add app hang reporting
+
+App hangs are an iOS-specific type of error that happens when the application is unresponsive for too long. App hangs are reported through the iOS SDK (not through [Logs][10]). By default, app hangs reporting is **disabled**, but you can enable it and set your own threshold to monitor app hangs that last longer than a duration you can specify in the `appHangThreshold` initialization parameter. Wh ...
```
```suggestion
App hangs are an iOS-specific type of error that happens when the application is unresponsive for too long. App hangs are reported through the iOS SDK (not through [Logs][10]). By default, app hangs reporting is **disabled**, but you can enable it and set your own threshold to monitor app hangs that last longer than a duration you can specify in the `appHangThreshold` initialization parameter. When enabled, any main thread pause that is longer than the specified `appHangThreshold` is considered a "hang" in [**Error Tracking**][1]. A customizable threshold allows you to find the right balance between fine-grained and noisy observability. See [Configure the app hang threshold](#configure-app-hang-threshold) for more guidance on setting this value.
```
Just noticed that we use `"Crash"` later on, but I know this is a super nitpick!

janine-c commented on content/en/error_tracking/frontend/mobile/ios.md:
```diff
@@ -1,8 +1,746 @@
+
+**Note:** Error Tracking supports symbolication of system symbol files for iOS v14+ arm64 and arm64e architecture. `.dSYM` files are limited in size to **2 GB** each.
+
+To help you debug errors, Datadog uses a unique generated build ID to deobfuscated stack traces, matching them with their corresponding mapping files. This process occurs regardless of whether the mapping files were uploaded during pre-production or production builds, ensuring the correct information is available for efficient QA processes when reviewing crashes and errors in Datadog.
```
```suggestion
To help you debug errors, Datadog uses a unique generated build ID to deobfuscate stack traces, matching them with their corresponding mapping files. This process occurs regardless of whether the mapping files were uploaded during pre-production or production builds, ensuring the correct information is available for efficient QA processes when reviewing crashes and errors in Datadog.
```

janine-c commented on content/en/error_tracking/frontend/mobile/ios.md:
```diff
@@ -1,8 +1,746 @@
+
+{{% /collapse-content %}}
+
+{{% collapse-content title="Add watchdog termination reporting" level="h4" expanded=false id="add-watchdog-termination-reporting" %}}
```
I just noticed that we use both `termination reporting` and `terminations reporting` in this section. Might be worth choosing one and sticking with it? Unless we use the plural in the UI somewhere, I think `termination reporting` sounds a smidge less awkward to me, but it's totally up to you.

rtrieu replied:
thanks for noticing this! after a google search, i determined "watchdog terminations reporting" would be the most appropriate here.

janine-c commented on content/en/real_user_monitoring/application_monitoring/ios/setup.md:
```diff
@@ -98,50 +104,28 @@ DatadogRUM.xcframework
-The SDK should be initialized as early as possible in the app lifecycle, specifically in the `AppDelegate`'s `application(_:didFinishLaunchingWithOptions:)` callback. This ensures all measurements, including application startup duration, are captured correctly. For apps built with SwiftUI, you can use `@UIApplicationDelegateAdaptor` to hook into the `AppDelegate`.
+The SDK should be initialized as early as possible in the app lifecycle, specifically in the `AppDelegate`'s `application(_:didFinishLaunchingWithOptions:)` callback. The `AppDelegate` is your app's main entry point that handles app lifecycle events. 
+
+This ensures all measurements, including application startup duration, are captured correctly. For apps built with SwiftUI, you can use `@UIApplicationDelegateAdaptor` to hook into the `AppDelegate`.
```
```suggestion
This ensures the SDK can correctly capture all measurements, including application startup duration. For apps built with SwiftUI, you can use `@UIApplicationDelegateAdaptor` to hook into the `AppDelegate`.
```
Not sure if this is right; I was just trying to use active voice. Ignore me if this makes less sense!

rtrieu commented on content/en/real_user_monitoring/application_monitoring/ios/setup.md:
```diff
@@ -98,50 +104,28 @@ DatadogRUM.xcframework
+3. To instrument your web views, click the **Instrument your webviews** toggle. For more information, see [Web View Tracking][11].
+4. To disable automatic user data collection for client IP or geolocation data, use the toggles for those settings. For more information, see [RUM iOS Data Collected][12].
 
    {{< img src="real_user_monitoring/ios/ios-create-application.png" alt="Create a RUM application for iOS in Datadog" style="width:100%;border:none" >}}
```
fixed!
